plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/procedure/IcebergOptimizeHandle.java

```diff
@@ -88,4 +91,10 @@ public DataSize getMaxScannedFileSize()
     {
         return maxScannedFileSize;
     }
+
+    @JsonProperty
+    public boolean isRetriesEnabled()
```

Comment:
could the RetryMode be provided by the engine to the `finish*()` methods, so that we don't have to embed the info in handles?

Reply:
We could - but I would opt for what we do now. It feels more natural - as we know information upfront. Also it allows for rejecting the request sooner if given retry mode is not supported by a connector.

Reply:
> allows for rejecting the request sooner if given retry mode is not supported by a connector.

i did not suggest not to provide it to begin methods.

Reply:
I would be asymetric vs other "stuff" we pass to "begin*" methods (layout, list of columns). I will leave it as is.